src/diffusers/models/open_muse_transformer.py

```diff
@@ -0,0 +1,508 @@
+        )
+
+        self.attention_blocks = nn.ModuleList(
+            [
+                nn.ModuleDict(
+                    dict(
+                        **(
+                            {"kv_mapper": nn.Linear(hidden_size, channels, use_bias)}
+                            if hidden_size != channels
+                            else {}
+                        ),
+                        attn=BasicTransformerBlock(
+                            channels,
+                            block_num_heads,
+                            channels // block_num_heads,
+                            attention_dropout,
+                            channels,
+                            attention_bias=use_bias,
+                            only_cross_attention=True,
+                            attention_out_bias=use_bias,
+                            norm_type="rms_norm",
+                            skip_ff=True,
+                            norm_elementwise_affine=ln_elementwise_affine,
+                            norm_eps=layer_norm_eps,
+                        ),
+                    )
+                )
+                for _ in range(num_res_blocks)
+            ]
+        )
+
+        if upsample:
+            self.upsample = nn.ModuleList(
+                [
+                    RMSNorm(channels, layer_norm_eps, ln_elementwise_affine),
+                    nn.ConvTranspose2d(channels, channels, kernel_size=2, stride=2, bias=use_bias),
+                ]
+            )
+        else:
+            self.upsample = None
+
+    def forward(self, x, cond_embeds, encoder_hidden_states):
+        if self.downsample is not None:
+            x = self.downsample[0](x.permute(0, 2, 3, 1)).permute(0, 3, 1, 2)
+            x = self.downsample[1](x)
+
+        for res_block, attention_block in zip(self.res_blocks, self.attention_blocks):
+            x_res = x
+            x = res_block["depthwise"](x)
+            x = res_block["norm"](x.permute(0, 2, 3, 1))
+            x = res_block["channelwise"](x).permute(0, 3, 1, 2)
+            x = x + x_res
+
+            scale, shift = res_block["cond_embeds_mapper"](F.silu(cond_embeds)).chunk(2, dim=1)
+            x = x * (1 + scale[:, :, None, None]) + shift[:, :, None, None]
+
+            if "kv_mapper" in attention_block:
+                encoder_hidden_states_ = attention_block["kv_mapper"](F.silu(encoder_hidden_states))
+            else:
+                encoder_hidden_states_ = encoder_hidden_states
+
+            batch_size, channels, height, width = x.shape
+            x = x.view(batch_size, channels, height * width).permute(0, 2, 1)
+            x = attention_block["attn"](x, encoder_hidden_states=encoder_hidden_states_)
+            x = x.permute(0, 2, 1).view(batch_size, channels, height, width)
+
+        if self.upsample is not None:
+            x = self.upsample[0](x.permute(0, 2, 3, 1)).permute(0, 3, 1, 2)
+            x = self.upsample[1](x)
+
+        return x
+
+
+class ConvMlmLayer(nn.Module):
```

Comment:
Could use a better name, I'm not sure what Mlm means

Reply:
Masked language model.

Reply:
Yeah I don't really see anything in here language model specific, so I think we should either have a different name for the block or just put it in the main transformer as a module dict

Reply:
just going to leave, this name is good enough on second thought